collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py

```diff
@@ -136,16 +135,14 @@ def __getitem__(self, idx, min_doc_length=16):
         # [CLS] <document a> [SEP] <document b> [SEP]
         num_special_tokens = 3
 
-        # TODO: Make seq_length = 512 for the last 10% of epochs, as specified
-        # in BERT paper
-        seq_length = self.max_seq_length - num_special_tokens
-
-        a_length = random.randrange(min_doc_length,
-                                    seq_length - min_doc_length + 1)
-        b_length = seq_length - a_length
+        max_num_tokens = self.max_seq_length - num_special_tokens
+        target_seq_length = max_num_tokens
+        if random.random() < self.short_seq_prob:
+            target_seq_length = random.randint(2, max_num_tokens)
```

Comment:
1. 2 is really small. compared to 512, it'll increase the number of batches by 256. shouldn't we make the smallest value a bit bigger, like 128? we can also random from a normal distribution like N(256, 10) instead of a uniform.

2. can we still keep the comment TODO: make seq_length = 512 for the last 10% of epochs?

Reply:
`min_num=2` and `randint` was following the original BERT implementation:
https://github.com/google-research/bert/blob/master/create_pretraining_data.py#L241

But that's a good idea to give users as argument to control, so we can experiment with.
I'll put it as comment.

Re. 3.: That's 2-phase pre-training for speed up and is controlled outside of this.